src/components/Accordion/Accordion.js

```diff
@@ -44,28 +55,43 @@ class Accordion extends Component {
 
   constructor(props) {
     super(props);
-    const { open } = props;
-    if (open === null) {
-      this.state = { activeSection: props.defaultOpen ? props.defaultOpen : props.children[0].props.id };
+    const { children, defaultOpen, open } = props;
+    if (open === null) { // controlled
+      if (defaultOpen) {
+        if (typeof defaultOpen === 'string') {
+          this.state = { activeSections: [defaultOpen] };
+        } else { // is array of strings
+          this.state = { activeSections: defaultOpen };
+        }
+      } else {
+        this.state = { activeSections: [children[0].props.id] };
+      }
     }
   }
 
   handleClick = (id) => {
-    const { open, onSectionClick } = this.props;
-    if (open === null) {
-      this.setState({ activeSection: id });
-    } else {
-      onSectionClick(id);
+    const { open, onSectionClick, multiple } = this.props;
+    if (open !== null) { // controlled
+      onSectionClick(id); return;
     }
+    // uncontrolled
+    this.setState((prevState) => {
+      if (!prevState.activeSections.includes(id)) {
+        if (multiple) return { activeSections: [...prevState.activeSections, id] };
+        return { activeSections: [id] };
+      }
+      return { activeSections: prevState.activeSections.filter(a => a !== id) };
```

Comment:
Instead of accessing `prevState.activeSections` multiple times, you could destructure the `prevState` argument like `this.setState(({ activeSections: prevActiveSections }) => {})`.